pages/api/book-a-visit.js

```diff
@@ -103,14 +81,22 @@ export default withContainer(
     }
 
     try {
-      let callId = ids.generate();
       let { wardId, trustId } = userIsAuthenticatedResponse;
-      let callPassword = ids.generate();
 
-      if (process.env.ENABLE_WHEREBY == "yes") {
-        callId = await wherebyCallId(body.callTime);
+      const { trust, error: trustErr } = await container.getRetrieveTrustById()(
+        trustId
+      );
+      if (trustErr) {
+        throw trustErr;
       }
 
+      const callIdProvider = new CallIdProvider(
+        trust.videoProvider,
+        body.callTime
+      );
+      const callId = await callIdProvider.generate();
+      let callPassword = ids.generate();
```

Comment:
I really like dis - how you've encapsulated this into a class with a switcharoo of providers.